Mathlib/Algebra/EuclideanDomain/Basic.lean

```diff
@@ -318,6 +318,75 @@ theorem mul_div_mul_comm_of_dvd_dvd {a b c d : R} (hac : c ∣ a) (hbd : d ∣ b
   rw [mul_div_cancel_left₀ _ hc0, mul_div_cancel_left₀ _ hd0, mul_mul_mul_comm,
     mul_div_cancel_left₀ _ (mul_ne_zero hc0 hd0)]
 
+theorem div_add_of_dvd {x y z : R} (h1 : y ≠ 0) (h2 : y ∣ x) : x / y + z = (x + y * z) / y := by
+  apply eq_div_of_mul_eq_right h1
+  rw [mul_add]
+  rw [EuclideanDomain.mul_div_cancel' h1 h2]
+
+theorem div_sub_of_dvd {x y z : R} (h1 : y ≠ 0) (h2 : y ∣ x) : x / y - z = (x - y * z) / y := by
+  apply eq_div_of_mul_eq_right h1
+  rw [mul_sub]
+  rw [EuclideanDomain.mul_div_cancel' h1 h2]
+
+theorem add_div_of_dvd {x y z : R} (h1 : z ≠ 0) (h2 : z ∣ y) : x + y / z = (z * x + y) / z := by
+  apply eq_div_of_mul_eq_right h1
+  rw [mul_add]
+  rw [EuclideanDomain.mul_div_cancel' h1 h2]
+
+theorem sub_div_of_dvd {x y z : R} (h1 : z ≠ 0) (h2 : z ∣ y) : x - y / z = (z * x - y) / z:= by
+  apply eq_div_of_mul_eq_right h1
+  rw [mul_sub]
+  rw [EuclideanDomain.mul_div_cancel' h1 h2]
+
+theorem div_div {x y z : R} (h1 : y ≠ 0) (h2 : z ≠ 0) (h3 : y ∣ x) (h4 : z ∣ (x / y)) :
```

Comment:
This may be true in the case that `y = 0` or `z = 0`?

Reply:
you are right.

good catch.